infra/Pos/Slotting/MemState/Class.hs

```diff
@@ -8,24 +8,38 @@ import           Control.Monad.Trans (MonadTrans)
 import           Universum
 
 import           Pos.Core.Types      (EpochIndex, Timestamp)
-import           Pos.Slotting.Types  (SlottingData)
+import           Pos.Slotting.Types  (EpochSlottingData)
 
 -- | 'MonadSlotsData' provides access to data necessary for slotting to work.
 class Monad m => MonadSlotsData m where
 
     getSystemStart :: m Timestamp
 
-    getSlottingData :: m SlottingData
+    getEpochLastIndex :: m EpochIndex
 
-    waitPenultEpochEquals :: EpochIndex -> m ()
+    -- TODO(ks): Maybe add other usefull functions directly?
```

Comment:
Well, ideally this class should be deleted and should become a simple reader over `SlottingVar` instead. But that's for future I think.